shell/common/api/electron_api_clipboard_mac.mm

```diff
@@ -19,6 +23,27 @@
   return GetFindPboardText();
 }
 
+std::vector<base::FilePath> Clipboard::ReadFilePaths() {
+  std::vector<base::FilePath> result;
+  NSArray* paths = [[NSPasteboard generalPasteboard]
+      propertyListForType:NSFilenamesPboardType];
```

Comment:
similar code in chromium wraps with `@try` and also checks `[pasteboard types]` before calling `propertyListForType:`: https://source.chromium.org/chromium/chromium/src/+/master:ui/base/clipboard/clipboard_mac.mm;l=465;drc=bdf1e0a5273ce828b48801de30f88965c76eb6f9

should we do the same?

Reply:
The referenced code is reading image which may throw, reading file paths is safe.

There is Chromium code reading file paths from drag drop clipboard:
https://source.chromium.org/chromium/chromium/src/+/master:ui/base/dragdrop/os_exchange_data_provider_mac.mm;l=244;drc=bdf1e0a5273ce828b48801de30f88965c76eb6f9?ss=chromium%2Fchromium%2Fsrc